schrack/schrack.pro

```diff
@@ -0,0 +1,20 @@
+include(../plugins.pri)
+
+QT += serialport serialbus
+
+SOURCES += \
+    integrationpluginschrack.cpp \
+    cionmodbusrtuconnection.cpp \
+    ../modbus/modbusdatautils.cpp
+
+HEADERS += \
+    integrationpluginschrack.h \
+    cionmodbusrtuconnection.h \
+    ../modbus/modbusdatautils.h
+
+OTHER_FILES += cion-registers.json
+
+modbusconnection.commands = python $${top_srcdir}/modbus/tools/generate-connection.py -j $${_PRO_FILE_PWD_}/cion-registers.json -o $${_PRO_FILE_PWD_} -c CionModbusRtuConnection
+QMAKE_EXTRA_TARGETS += modbusconnection
+
+#target.depends += modbusconnection
```

Comment:
Nice, I would not auto generate the classes on every build, so commenting this out is good. The only reason why I don't wanna generate the classes during build is that the tools changes over time, and you might want to migration of the new class individually. Once the tool is quiet stable we might wanna consider that.

Reply:
Yes... that's why I removed it. Also I ran into another problem with autogenerating which would need to be solved first but I can't recall atm what it was. Probably build-dependencies to python and such...

In any case, having the option to `make modbusconnection` to update everything is nice, I've been using it all the time. Eventually we should move that to the .pri so all modbus plugins have it available right away

Reply:
Right, so the plan is any way to move the modbus folder into a lib, and then also move the tool to a own dev package.